lib/api/core/models/repositories/profileRepository.js

```diff
@@ -271,6 +271,11 @@ class ProfileRepository extends Repository {
 
     return profile.validateDefinition()
       .then(() => {
+        if (profile._id === 'anonymous' && profile.policies.map(policy => policy.roleId).indexOf('anonymous') === -1) {
```

Comment:
No it cannot. The `validateDefinition` call checks the profile is well formed, and notably that profile.policies is a non-empty array, which is used in the map call.